server/cliserver_test.go

```diff
@@ -213,18 +212,6 @@ func TestGetEthChainID(t *testing.T) {
 	require := require.New(t)
 	assert := assert.New(t)
 
-	// test offchain
-	srv := newMockServer()
-	res, err := http.Get(fmt.Sprintf("%s/EthChainID", srv.URL))
-	require.Nil(err)
-	assert.Equal(http.StatusOK, res.StatusCode)
-	body, err := ioutil.ReadAll(res.Body)
-	require.Nil(err)
-	assert.Equal("0", string(body))
```

Comment:
Should we keep this test case?

Reply:
Do you mean the whole `cliserver_test.go`, right? It's kind-of integration/e2e test for the same what we test in `handlers_test.go`. I see some value in it, because if we, e.g., break something with the endpoints themselves, or in the REST server configuration of Livepeer, we would catch it here.

Reply:
I specifically mean the code here under `// test offchain` which appears to be a test case for when the `/EthChainID` endpoint should return 0. I see that we kept the test case starting on L228 for when the `/EthChainID` endpoint returns 1. So, I am wondering if we should have both?

Reply:
ahh, I see you point. No, I think we don't need it anymore. This test was actually in some sense wrong before. Because it tested the following scenario:

If [DB is nil](https://github.com/livepeer/go-livepeer/blob/master/server/cliserver_test.go#L29), then [return `0` from `/EthChainID`](https://github.com/livepeer/go-livepeer/blob/master/server/cliserver_test.go#L223).

But it's not a valid scenario, we never have a situation when DB is nil, we always create DB, [even for the off-chain mode](https://github.com/livepeer/go-livepeer/blob/master/cmd/livepeer/livepeer.go#L294). So, in result `// test offchain` is exactly the same as `// test onchain` in this test case. That's why I removed it and consolidated it into one.

And the root cause why I actually removed it was that now I check in the handler [if we have the DB object](https://github.com/livepeer/go-livepeer/blob/59a9437dcc83f956a23580bb4290ec6b1085eb7f/server/handlers.go#L63), so this test started to fail.

Reply:
Got it thanks for the clarification!